x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/BinaryDvConfirmedAutomatonQuery.java

```diff
@@ -27,32 +30,65 @@
 import java.util.Objects;
 
 /**
- * Query that runs an Automaton across all binary doc values.
- * Expensive to run so normally used in conjunction with more selective query clauses.
+ * Query that runs an Automaton across all binary doc values (but only for docs that also 
+ * match a provided approximation query which is key to getting good performance).
  */
-public class AutomatonQueryOnBinaryDv extends Query {
+public class BinaryDvConfirmedAutomatonQuery extends Query {
 
     private final String field;
     private final String matchPattern;
     private final ByteRunAutomaton bytesMatcher;
+    private final Query approxQuery;
 
-    public AutomatonQueryOnBinaryDv(String field, String matchPattern, Automaton automaton) {
+    public BinaryDvConfirmedAutomatonQuery(Query approximation, String field, String matchPattern, Automaton automaton) {
+        this.approxQuery = approximation;
         this.field = field;
         this.matchPattern = matchPattern;
         bytesMatcher = new ByteRunAutomaton(automaton);
     }
+    public BinaryDvConfirmedAutomatonQuery(String field, String matchPattern, Automaton automaton) {
```

Comment:
Mostly a personal preference, but I think it'd be clearer to remove this constructor and force the caller to pass in `MatchAllDocsQuery`. That makes it really clear to the caller this query might do a full scan!

Reply:
I did have it that way before and felt like I was repeating the same code but, agreed, it's probably safer to reiterate the match-all point.